common/unsafe/src/main/java/org/apache/spark/sql/catalyst/util/CollationAwareUTF8String.java

```diff
@@ -183,6 +204,19 @@ public static int findInSet(final UTF8String match, final UTF8String set, int co
     return 0;
   }
 
+  public static int lowercaseIndexOf(final UTF8String target, final UTF8String pattern,
```

Comment:
Please add documentation comments to all the functions, specifically regarding the indices (1-based) and the return values (here for example we want to say that in case we don't have a match we return `-1`).

Reply:
Added